syntax = "proto3";

package grpc_example;

option java_package = "com.example.grpc";
option java_outer_classname = "ServiceProto";


message RequestMessage {
  string name = 1;
}


message ResponseMessage {
  string message = 1;
}
message WeatherRequest {
  string city = 1;
}

message WeatherResponse {
  string update = 1;
}


service MyService {
  rpc GetResponse(RequestMessage) returns (ResponseMessage);
  rpc GetWeatherUpdates(WeatherRequest) returns (stream WeatherResponse);
}
